app/main.py

```diff
@@ -1,34 +1,128 @@
+from typing import List
+
+
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self,
+                 row: int,
+                 column: int,
+                 is_alive: bool = True
+                 ) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
+
+    def __repr__(self) -> str:
+        return f"Alive: {self.is_alive}"
+
+    def get_coordinate_in_tuple(self) -> tuple:
+        return self.row, self.column
+
+    def __eq__(self, other: tuple) -> bool:
+        return (self.row, self.column) == other
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self,
+                 start: tuple,
+                 end: tuple,
+                 is_drowned: bool = False) -> None:
+        self.start = start
+        self.end = end
+        self.is_drowned = is_drowned
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+        if ((self.start[0] == self.end[0])
+                and (self.start[1] == self.end[1])):
+            # it means 1 - size ship
+            self.decks = [Deck(row=self.start[0],
+                               column=self.start[1])]
+            # COULD BE A MISTAKE
+        elif self.start[0] == self.end[0]:
+            # horizontal ship
+            self.decks = [
+                Deck(
+                    row=self.start[0],
+                    column=column
+                )
+                for column in range(self.start[1], self.end[1] + 1)
+            ]
+            # + 1 to include the last coordinate (3)
+        elif self.start[1] == self.end[1]:
+            self.decks = [
+                Deck(
+                    row=row,
+                    column=self.start[1]
+                )
+                for row in range(self.start[0], self.end[0] + 1)
+            ]
```

Comment:
You can just use loop inside loop.
for row_index in range(coords based of start and end):
    for column_index in range(coords based of start and end):